src/Entity/UrlButton.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Entity\UrlButton.
+ */
+
+namespace Drupal\url_embed\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\url_embed\UrlButtonInterface;
+
+/**
+ * Defines the UrlButton entity.
+ *
+ * @ConfigEntityType(
+ *   id = "url_button",
+ *   label = @Translation("Url Button"),
+ *   handlers = {
+ *     "list_builder" = "Drupal\url_embed\UrlButtonListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\url_embed\Form\UrlButtonForm",
+ *       "edit" = "Drupal\url_embed\Form\UrlButtonForm",
+ *       "delete" = "Drupal\Core\Entity\EntityDeleteForm"
+ *     }
+ *   },
+ *   config_prefix = "url_button",
+ *   admin_permission = "administer url buttons",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *   },
+ *   config_export = {
+ *     "label",
+ *     "id",
+ *     "button_icon_uuid",
+ *   },
+ *   links = {
+ *     "edit-form" = "/admin/config/content/url-button/{url_button}",
+ *     "delete-form" = "/admin/config/content/url-button/{url_button}/delete"
+ *   }
+ * )
+ */
+class UrlButton extends ConfigEntityBase implements UrlButtonInterface {
+
+  /**
+   * The UrlButton ID.
+   *
+   * @var string
+   */
+  public $id;
+
+  /**
+   * Label of UrlButton.
+   *
+   * @var string
+   */
+  public $label;
+
+  /**
+   * UUID of the button's icon fili.
+   *
+   * @var string
+   */
+  public $button_icon_uuid;
+
+  /**
+   * Array of allowed display plugins for the URL.
+   *
+   * An empty array signifies that all are allowed.
+   *
+   * @var array
+   */
+  public $display_plugins;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSource(){
```

Comment:
Remove these functions which are not required for the interface. Can add back later, if required.